cg/services/delivery_message/utils.py

```diff
@@ -87,3 +89,11 @@ def validate_cases(cases: list[Case], case_ids: list[str]) -> None:
 
 def is_matching_order(cases: list[Case]) -> bool:
     return all([case.latest_order == cases[0].latest_order for case in cases])
+
+
+def get_cases_ready_for_delivery(trailblazer_api: TrailblazerAPI, order: Order) -> list[Case]:
```

Comment:
Design: this logic could be placed in a separate service wrapping around the trailblazer API client. Consider refactoring if you think it is worth it

Reply:
Similar to the housekeeper storage functions? @seallard

Reply:
Yes, I don't like those either. They probably belong in an `AnalysisFileService` or something

Reply:
But out of scope, if you think it makes sense and feel like doing it - extract it 👍

Reply:
I will think about it and implement in another PR!